### PR TITLE
Add CountParallel method

### PR DESCRIPTION
**Pull request by yyoshiki41**
hi, guy :sunglasses: 

this p-r creates `CountParallelWithCondition` method.

### CODE REVIEW

**Comment by yyoshiki41** [comment]
and, implemented common func which builds `FindCondition`.

**Comment by evalphobia** [comment]
`BuildFindCondtion` has many args, I think it could be reduced.
how about the method signature like below,

``` go
func (xpr *XormParallel) CreateSessionsWithCondition(cond FindCondition) []Session {
    slaves := xpr.orm.Slaves(cond.Table)
    var sessions []session

    // ... create sessions with condition

    return sessions
} 
```

**Comment by yyoshiki41** [comment]
Thanks for your review :+1: 
Removed `func BuildFindCondtion()` and
implemented `CreateSessionsWithCondition` method in `xorm_parallel.go`

File: orm/xorm/xorm_parallel.go [hunk 1]
```diff
@@ -76,3 +83,42 @@ func (xpr *XormParallel) FindParallelByCondition(listPtr interface{}, cond FindC
...
+		return nil, errors.NewErrArgType("objPtr must be a pointer")
+	}
+
+	// create session with the condition
+	slaves := xpr.orm.Slaves(cond.Table)
+	sessions := xpr.CreateSessionsWithCondition(cond)
+
+	// execute query
+	var errList []error
+	results := make(chan int64, len(slaves))
```
Comment (evalphobia) [inline orm/xorm/xorm_parallel.go]: @yyoshiki41  `len(slaves)` correspond to `len(sessions)`, so don't need to use `slaves` in this method.

**Comment by yyoshiki41** [comment]
@evalphobia 
Refactored `xorm_parallel.go`.
pls, review again..

**Comment by evalphobia** [comment]
:+1:

**Comment by yyoshiki41** [comment]
thx!!

**Pull request merged**